php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -746,11 +750,15 @@ class NDB_BVL_Instrument extends NDB_Page
             $values = Utility::nullifyEmpty($values, $key);
         }
         $db = \Database::singleton();
-        $db->update(
-            $this->table,
-            $values,
-            array('CommentID' => $this->getCommentID())
-        );
+        if ($this->jsonData !== true) {
```

Comment:
I would suggest putting this in a `try{}catch` rather then bypassing completely until everything is JSON ready (ie CouchDB scripts, get/set dataEntryCompletionStatus, all all other getters in the class still referring to $this->testname)

Reply:
I considered that when I was writing it, but I don't think a try/catch block is a good idea. It forces a query that's known to fail to be run for people who don't want it and will hide issues/errors from places that are supposed to be changed, rather than complaining loudly so that issues are caught right away.